docs/reference/indices/shrink-index.asciidoc

```diff
@@ -60,15 +54,44 @@ with the <<cat-recovery,`_cat recovery` API>>, or the <<cluster-health,
 `cluster health` API>> can be used to wait until all shards have relocated
 with the `wait_for_no_relocating_shards` parameter.
 
-[float]
-==== Shrinking an index
+
+[[shrink-index-api-desc]]
+==== {api-description-title}
+
+The shrink index API allows you to shrink an existing index into a new index
+with fewer primary shards. The requested number of primary shards in the target index
+must be a factor of the number of shards in the source index. For example an index with
+`8` primary shards can be shrunk into `4`, `2` or `1` primary shards or an index
+with `15` primary shards can be shrunk into `5`, `3` or `1`. If the number
+of shards in the index is a prime number it can only be shrunk into a single
+primary shard. Before shrinking, a (primary or replica) copy of every shard
+in the index must be present on the same node.
+
+Shrinking works as follows:
+
+* First, it creates a new target index with the same definition as the source
+  index, but with a smaller number of primary shards.
+
+* Then it hard-links segments from the source index into the target index. (If
+  the file system doesn't support hard-linking, then all segments are copied
+  into the new index, which is a much more time consuming process. Also if using 
+  multiple data paths, shards on different data paths require a full copy of 
+  segment files if they are not on the same disk since hardlinks don’t work across
+  disks)
+
+* Finally, it recovers the target index as though it were a closed index which
+  had just been re-opened.
```

Comment:
Referring to "Shrinking" as "it" strikes me as odd. Maybe:

A shrink operation:

1. Creates the target index...
2. Hard links...
3. Recovers the target index...

Reply:
Fixed with e6f9b25b736.